src/normal.cc

```diff
@@ -553,7 +559,8 @@ void pipe(Context& context, NormalParams params)
         prompt, {}, default_command, context.faces()["Prompt"],
         PromptFlags::DropHistoryEntriesWithBlankPrefix, '|',
         shell_complete,
-        [default_command](StringView cmdline, PromptEvent event, Context& context)
+        [default_command, selection_edition=std::make_shared<ScopedSelectionEdition>(context)]
```

Comment:
~That make_shared looks suspicious, I guess that means we end-up copying this lambda around, is that something we can avoid ?~

Looked it up, this is due to std::function requiring a copyiable target, isn't it... Well maybe its going to be time to replace that with our own, or to use std::move_only_function once we migrate to C++23. I guess for now we can live with the shared_ptr here.

Reply:
yeah the commit message tries to mention that but didn't give enough detail.
We could backport std::move_only_function (or perhaps add a template parameter to Prompt but that's probably not worth it)